src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/rca/hotheap/HighHeapUsageYoungGenRca.java

```diff
@@ -118,6 +133,67 @@ protected void remove(SlidingWindowData e) {
     this(rcaPeriod, 1.0, heap_Used, gc_Collection_Time);
   }
 
+  private boolean fullGcTimeTooHigh(double avgFullGcTime) {
+    return (!Double.isNaN(avgFullGcTime) && avgFullGcTime > FOLLOWER_CHECK_TIMEOUT_MS);
+  }
+
+  private boolean promotionRateTooHigh(double avgPromotionRate, double modifier) {
```

Comment:
Minor: is the `modifier` intended to change the threshold? if so, can we just override the threshold in an overloaded fn call?

Reply:
This is an artifact of a refactor of the operate() method. I only included the modifier param to be able to factor out this logic. Since this isn't related to code I wrote, I'm going to leave this as is for now.